### ISSUE TITLE
[QUESTION] DbRefs

### ISSUE THREAD

**Issue by belius** [litedb-org/LiteDB#2080]
Hi,

I am working on a project where I have an object with a list of other objects.

I would like to use the DbRef feature, but the problem is that the list of other objects is a List of dynamics and each object in there is not a typed object (could be a JObject or a Dictionary of string/object)

BTW I can know per each dynamic the name of the collection where it is stored.

Is it possible to use DbRefs in this scenario, and in case how?

I would need this to implement a sort of many-to-many between different objects that share certain data

thanks

**Comment by JKamsker** [comment]
Yes. A DbRef is just a document `{ "$id": <id>, "$ref": "<collection>" }`, and `Include` resolves each one from the collection named in its own `$ref`. Include doesn't depend on static types, so a list of untyped items that point to different collections works:

```csharp
static BsonDocument Ref(BsonValue id, string collection) =>
    new BsonDocument { ["$id"] = id, ["$ref"] = collection };

public class Parent
{
    public int Id { get; set; }
    public List<BsonDocument> Items { get; set; }
}

parents.Insert(new Parent { Id = 1, Items = new List<BsonDocument> { Ref(1, "tags"), Ref("joe", "people") } });

var p = parents.Include(BsonExpression.Create("$.Items[*]")).FindById(1);
// Items: {"$id":1,"label":"red"}, {"$id":"joe","name":"Joe"}
```

It works fully untyped too, with `db.GetCollection("parents").Include("$.Items[*]")`, or in SQL: `SELECT $ FROM parents INCLUDE $.Items[*]`. The referenced document replaces the `{ $id, $ref }` entry, and its `_id` shows up as `$id`. If your items are `Dictionary<string, object>`, `BsonMapper.Global.Serialize(dict).AsDocument` turns them into a `BsonDocument`.

`[BsonRef]` on typed properties only automates writing these reference documents, so writing them by hand like this is fine.

Closing this as answered — if it doesn't cover your case, comment here and we'll reopen it.
